pandas/core/computation/expr.py

```diff
@@ -39,10 +36,125 @@
 
 import pandas.io.formats.printing as printing
 
+# A token value Python's tokenizer probably will never use.
+_BACKTICK_QUOTED_STRING = 100
+
+Tok = Tuple[int, str]
+
+
+def _create_valid_python_identifier(name: str) -> str:
+    """Create valid Python identifiers from any string.
+
+    Check if name contains any special characters. If it contains any
+    special characters, the special characters will be replaced by a special
+    string and a prefix is added.
+
+    Raises
+    ------
+    If the returned name is not a Python valid identifier, raise an exception.
 
-def tokenize_string(source):
     """
-    Tokenize a Python source code string.
+    if name.isidentifier() and not iskeyword(name):
+        return name
+
+    # Create a dict with the special characters and their replacement.
+    # To create the dict, we loop over all EXACT_TOKEN_TYPES with a single character
+    # (Multiple character operators will be replaced as well because they are
+    # composed of single character operators.)
+    # and take the character alongside with its human readable description from
+    # token.tok_name.
+    # This description is surrounded with underscores for better
+    # readability in the final name.
+    # Also added some more characters, that are not operators,
+    # that people might use in column names.
+    special_characters_replacements = {
+        " ": "_",
+        "?": "_QUESTIONMARK_",
+        "!": "_EXCLAMATIONMARK_",
+        "$": "_DOLLARSIGN_",
+        "€": "_EUROSIGN_",
+        # Including quotes works, but they have to come in pairs.
+        "'": "_SINGLEQUOTE_",
+        '"': "_DOUBLEQUOTE_",
+        # Currently not possible. Terminates parser and won't find backtick.
+        # "#": "_HASH_",
+        **{
+            char: "_" + token.tok_name[tokval] + "_"
+            # The ignore here is because of a bug in mypy that is resolved in 0.720
+            for char, tokval in tokenize.EXACT_TOKEN_TYPES.items()  # type: ignore
+            if len(char) == 1
```

Comment:
Hmm I thought from a previous review we didn't need to limit this to just one character?

Reply:
Well, the point is, multiple character operators are automatically replaced by single character operators. So multiple character operators *will* be replaced. However, for performance reasons, we don't have to check the string for multiple character operators. So, I just remove them here, as they will be replaced anyway.

Reply:
Cool thanks for the clarification. Unless there is a tangible performance benefit would rather keep the implementation simple

Reply:
Well, there is also a factor of undefined behavior. Officially the ordering of the dict is not guaranteed in Python version < 3.7, which means that if there is a multiple character operator inside the string there are randomized two outcomes: Are the multiple character operator is replaced by the multiple character itself, or it is replaced by the individual characters it is made off. 

Very, very unlikely to go wrong. IIRC the CPython implementation already does it, but there are no guarantees, so I rather have this line in place instead.

Reply:
Doesn't the iteration on line 89 go character by character anyway? So wouldn't match on multi character keys

Reply:
You are right. I forgot how exactly I implemented the character replacement and made wrong assumptions about it. In that case, we can just remove this line.